lib/pbench/test/unit/server/query_apis/test_datasets_update.py

```diff
@@ -228,10 +228,134 @@ def fake_bulk(
+
+    @pytest.mark.parametrize(
+        "owner,access",
+        [
+            ("drb", "public"),
+            ("drb", "private"),
+            ("test", "public"),
+            ("test", "private"),
+        ],
+    )
+    def test_query_owner_publish(
+        self,
+        access,
+        attach_dataset,
+        build_auth_header,
+        client,
+        create_drb_user,
+        create_user,
+        get_document_map,
+        monkeypatch,
+        owner,
+        server_config,
+    ):
+        """
+        Check behavior of the datasets_update API with various combinations of dataset
+        access and owner (managed by the "owner" parametrization here) and
+        authenticated user (managed by the build_auth_header fixture).
+        """
+        self.fake_elastic(monkeypatch, get_document_map, False)
+
+        is_admin = build_auth_header["header_param"] == HeaderTypes.VALID_ADMIN
+        if not HeaderTypes.is_valid(build_auth_header["header_param"]):
+            expected_status = HTTPStatus.UNAUTHORIZED
+        elif not is_admin:
+            expected_status = HTTPStatus.FORBIDDEN
+        else:
+            expected_status = HTTPStatus.OK
+
+        ds = Dataset.query(name=owner)
+        assert_id = str(create_drb_user.id) if owner == "drb" else str(create_user.id)
+        response = client.post(
+            f"{server_config.rest_uri}/datasets/{ds.resource_id}",
+            headers=build_auth_header["header"],
+            query_string={"owner": owner, "access": access},
+        )
+
+        assert response.status_code == expected_status
+        if expected_status == HTTPStatus.OK:
+            assert response.json == {"ok": 31, "failure": 0}
+            dataset = Dataset.query(name=owner)
+            assert dataset.access == access
+            assert dataset.owner_id == assert_id
+
+    def test_invalid_owner_params(
+        self,
+        client,
+        get_document_map,
+        monkeypatch,
+        pbench_admin_token,
+        server_config,
+    ):
+        """
+        Check the datasets_update API response if the "owner" attribute is invalid.
+        """
+
+        ds = Dataset.query(name="drb")
+        response = client.post(
+            f"{server_config.rest_uri}/datasets/{ds.resource_id}",
+            headers={"authorization": f"Bearer {pbench_admin_token}"},
+            query_string={"owner": str("invalid_owner")},
+        )
+
+        # Verify the report and status
+        assert response.status_code == HTTPStatus.NOT_FOUND
+        assert (
+            response.json["message"]
+            == "Value 'invalid_owner' (str) cannot be parsed as a username"
+        )
+
+    def test_notadmin_owner(
+        self,
+        client,
+        create_drb_user,
+        get_document_map,
+        monkeypatch,
+        pbench_token,
+        server_config,
+    ):
+        """
+        Check the datasets_update API response if the "owner" and "access" provided is valid.
+        """
```

Comment:
I don't think that docstring describes _this_ test.

Also, I think the name of this test could benefit from some tweaking:  the non-Admin here is the _requestor_, not the proposed new owner for the dataset.